### DISCUSSION TITLE
Ability to keep original filename ("download"/"public_url") in database and on disk

### DISCUSSION THREAD

**Discussion by luciandavidescu** [directus/directus#11351] [Feature Requests]
This was discussed multiple times (@ [#3014](https://github.com/directus/directus/discussions/3014), [#3182](https://github.com/directus/directus/discussions/3182), [#3279](https://github.com/directus/directus/discussions/3279), [#7612](https://github.com/directus/directus/discussions/7612), [#9079](https://github.com/directus/directus/discussions/9079), [#10575](https://github.com/directus/directus/discussions/10575)) and there's a pending pull request (@ [#8065](https://github.com/directus/directus/pull/8065)), but I'll try to add to those by make a more extensive case of why I think it's necessary:

### I. Deployment

a) **Static.** In a purely static (or "serverless" as the buzzword goes) setup there'd be no public access to the API at all. On the simplest of cases, someone spins a Directus instance on their laptop, writes a blogpost, commit to gh-pages and such and then turns the laptop off expecting everything to stay live`*`.

b) **CDN.** If you need to load-balance your assets from several servers, the DB and the API endpoint become a hurdle. If the focus is on the assets (especially with video), than it should be as easy as rsync them to yet another location and serve from there as well.

### II. Stability

a) **Concurrency.** A node process in nowhere nearly as performant as a specialized webserver in handling too many simultaneous requests, so for high-concurrency setups you may want to rely on a tried and tested webserver.

b) **Security.** If your machine is subject to DDOS, a dynamic resizing module may be one of the weak spots and you'd want to be able to disable it at least temporarily without degrading the service.

### III. Semantics

a) **Usability.** Some use the filename to quickly assess the content of images, to softly enforce credits/attribution or even programatically, to associate a resource directly (e.g username corresponds to filename) .

b) **SEO.** If the volume of incoming hits from image search is non-negligible, that bit of extra info that comes with the filename itself does help.

### IV. Portability

a) **Moving in.** If the old database was doing things one way, you'd either have to refactor legacy content or maintain it as such with the added complexity, and with the risk that someday it may silently fail.

b) **Moving out.** The other way around too – _keeping all your data pure, organized and portable, end-to-end_ would be of little use if linked resources are stored in a non-generic fashion.

---

`*` (My bet / use-case right now is that this approach is quite scalable, give or take a few issues.)

At first I thought there must certainly be a way to keep the original filename and was quite surprised to find out that it's actually not possible (started researching when for some reason the access_token got stripped and I saw that broken image icon).

The way things work is indeed nicely done, especially with the on-the-fly resizing... I thought _great, I'll use this to build a nice all-around srcset and leave the original in the src for reference and fallback_ (granted, browsers don't yet implement [the fallback requirement from the html spec](https://www.w3.org/TR/2021/NOTE-html53-20210128/semantics-embedded-content.html#elementdef-source ), but it is an official spec so they might implement it some day). 

As a quick fix for my use case, I'll most likely find a way to rewrite the filename both in the database and on disk, but that means leaving out the great resizing features, as per...

> While you may _technically_ be able to expose your storage adapters root filesystem and access your raw files through there, it is recommended that you always use the Directus API. This is the only way that you can take advantage of file permissions and other built-in features

...and also is something that a lot of people would expect to work by default, or as an easy checkable option.

**Comment by lsanchezs** [comment]
For us, it is very relevant for a better SEO.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
